libvoikko/typescript/libvoikko.d.ts

```diff
@@ -0,0 +1,198 @@
+export = Libvoikko;
+
+declare class Libvoikko {
+  constructor();
+  init(language: string): Voikko;
+}
+
+declare interface GrammarError {
+  errorCode: number;
+  startPos: number;
+  errorLen: number;
+  suggestions: Array<string>;
+  shortDescription: string;
+}
+
+declare interface Analysis {
+  BASEFORM: string;
+  CLASS: string;
+  FSTOUTPUT: string;
+  NUMBER: string;
+  SIJAMUOTO: string;
+  STRUCTURE: string;
+  WORDBASES: string;
```

Comment:
These attributes are returned for nouns. There are far more attributes that can be returned for example for verbs. On the other hand it is not guaranteed that any attributes are present (especially if analyzing languages other than Finnish). Perhaps you could take all attributes listed in https://github.com/voikko/corevoikko/blob/master/libvoikko/doc/morphological-analysis.txt but mark them optional?

Reply:
Sounds like a good plan. I'll add some documentation for those attributes.